lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -374,8 +374,25 @@ def linearized_ancestors_of(fully_qualified_name)
       cached_ancestors = @ancestors[fully_qualified_name]
       return cached_ancestors if cached_ancestors
 
+      parts = fully_qualified_name.split("::")
```

Comment:
`linearized_ancestors_of` is getting really long (149 lines). Should we consider extracting out some methods?

Reply:
I did my best in the latest commit. I split `linearize_mixins` and `linearize_superclass` into their own methods.